lib/state/appActions.js

```diff
@@ -236,16 +236,39 @@ appActions.login = function(credentials, options, cb) {
   });
 };
 
+//make sure we have the correct default targetId
+appActions._getDefaultTargetId = function(loggedInUser){
+  //default to the logged in users id
+  var userid = loggedInUser.userid;
+
+  //even though we have multiple groups there may still only be one upload account
+  if (_.isEmpty(loggedInUser.uploadGroups) === false || loggedInUser.uploadGroups.length > 1) {
```

Comment:
This is really confusing to read, and I'm still working out whether the logic is right. Wouldn't `!_.isEmpty(loggedInUser.uploadGroups)` (and no `||...`) accomplish the same thing? Because what you're asking here is `uploadGroups.length !== 0 || uploadGroups.length > 1`, which is just the same thing is `uploadGroups.length > 0`.